crates/typst-library/src/meta/counter.rs

```diff
@@ -301,8 +301,10 @@ impl Counter {
 
     /// The selector relevant for this counter's updates.
     fn selector(&self) -> Selector {
-        let mut selector =
-            Selector::Elem(UpdateElem::elem(), Some(dict! { "key" => self.0.clone() }));
+        let mut selector = Selector::Elem(
+            UpdateElem::elem(),
+            Some(fields! { UpdateElemFields::Key => self.0.clone() }),
```

Comment:
It could but that would inherently make the trait non object safe, which is why I opted for this technique, admittedly it could be private and in fact, its visibility is dependent on the element's visibility.